File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-asset-5-4/ExchangeTradedContract.cpp
// ExchangeTradedContract.cpp 
#include "ExchangeTradedContract.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ExchangeTradedContract::ExchangeTradedContract(TiXmlNode* xmlNode)
: ExchangeTraded(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //multiplierNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* multiplierNode = xmlNode->FirstChildElement("multiplier");

   if(multiplierNode){multiplierIsNull_ = false;}
   else{multiplierIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- multiplierNode , address : " << multiplierNode << std::endl;
   #endif
   if(multiplierNode)
   {
      if(multiplierNode->Attribute("href") || multiplierNode->Attribute("id"))
      {
          if(multiplierNode->Attribute("id"))
          {
             multiplierIDRef_ = multiplierNode->Attribute("id");
             multiplier_ = boost::shared_ptr<XsdTypePositiveInteger>(new XsdTypePositiveInteger(multiplierNode));
             multiplier_->setID(multiplierIDRef_);
             IDManager::instance().setID(multiplierIDRef_,multiplier_);
          }
          else if(multiplierNode->Attribute("href")) { multiplierIDRef_ = multiplierNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { multiplier_ = boost::shared_ptr<XsdTypePositiveInteger>(new XsdTypePositiveInteger(multiplierNode));}
   }

   //contractReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* contractReferenceNode = xmlNode->FirstChildElement("contractReference");

   if(contractReferenceNode){contractReferenceIsNull_ = false;}
   else{contractReferenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- contractReferenceNode , address : " << contractReferenceNode << std::endl;
   #endif
   if(contractReferenceNode)
   {
      if(contractReferenceNode->Attribute("href") || contractReferenceNode->Attribute("id"))
      {
          if(contractReferenceNode->Attribute("id"))
          {
             contractReferenceIDRef_ = contractReferenceNode->Attribute("id");
             contractReference_ = boost::shared_ptr<XsdTypeString>(new XsdTypeString(contractReferenceNode));
             contractReference_->setID(contractReferenceIDRef_);
             IDManager::instance().setID(contractReferenceIDRef_,contractReference_);
          }
          else if(contractReferenceNode->Attribute("href")) { contractReferenceIDRef_ = contractReferenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { contractReference_ = boost::shared_ptr<XsdTypeString>(new XsdTypeString(contractReferenceNode));}
   }

   //expirationDateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* expirationDateNode = xmlNode->FirstChildElement("expirationDate");

   if(expirationDateNode){expirationDateIsNull_ = false;}
   else{expirationDateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- expirationDateNode , address : " << expirationDateNode << std::endl;
   #endif
   if(expirationDateNode)
   {
      if(expirationDateNode->Attribute("href") || expirationDateNode->Attribute("id"))
      {
          if(expirationDateNode->Attribute("id"))
          {
             expirationDateIDRef_ = expirationDateNode->Attribute("id");
             expirationDate_ = boost::shared_ptr<AdjustableOrRelativeDate>(new AdjustableOrRelativeDate(expirationDateNode));
             expirationDate_->setID(expirationDateIDRef_);
             IDManager::instance().setID(expirationDateIDRef_,expirationDate_);
          }
          else if(expirationDateNode->Attribute("href")) { expirationDateIDRef_ = expirationDateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { expirationDate_ = boost::shared_ptr<AdjustableOrRelativeDate>(new AdjustableOrRelativeDate(expirationDateNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypePositiveInteger> ExchangeTradedContract::getMultiplier()
{
   if(!this->multiplierIsNull_){
        if(multiplierIDRef_ != ""){
             return boost::shared_static_cast<XsdTypePositiveInteger>(IDManager::instance().getID(multiplierIDRef_));
        }else{
             return this->multiplier_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypePositiveInteger>();
   }
}
boost::shared_ptr<XsdTypeString> ExchangeTradedContract::getContractReference()
{
   if(!this->contractReferenceIsNull_){
        if(contractReferenceIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeString>(IDManager::instance().getID(contractReferenceIDRef_));
        }else{
             return this->contractReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeString>();
   }
}
boost::shared_ptr<AdjustableOrRelativeDate> ExchangeTradedContract::getExpirationDate()
{
   if(!this->expirationDateIsNull_){
        if(expirationDateIDRef_ != ""){
             return boost::shared_static_cast<AdjustableOrRelativeDate>(IDManager::instance().getID(expirationDateIDRef_));
        }else{
             return this->expirationDate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<AdjustableOrRelativeDate>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-asset-5-4/ExchangeTradedContract.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ExchangeTradedContract : ExchangeTraded
    {
        public ExchangeTradedContract(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode multiplierNode = xmlNode.SelectSingleNode("multiplier");
            
            if (multiplierNode != null)
            {
                if (multiplierNode.Attributes["href"] != null || multiplierNode.Attributes["id"] != null) 
                {
                    if (multiplierNode.Attributes["id"] != null) 
                    {
                        multiplierIDRef_ = multiplierNode.Attributes["id"].Value;
                        XsdTypePositiveInteger ob = new XsdTypePositiveInteger(multiplierNode);
                        IDManager.SetID(multiplierIDRef_, ob);
                    }
                    else if (multiplierNode.Attributes["href"] != null)
                    {
                        multiplierIDRef_ = multiplierNode.Attributes["href"].Value;
                    }
                    else
                    {
                        multiplier_ = new XsdTypePositiveInteger(multiplierNode);
                    }
                }
                else
                {
                    multiplier_ = new XsdTypePositiveInteger(multiplierNode);
                }
            }
            
        
            XmlNode contractReferenceNode = xmlNode.SelectSingleNode("contractReference");
            
            if (contractReferenceNode != null)
            {
                if (contractReferenceNode.Attributes["href"] != null || contractReferenceNode.Attributes["id"] != null) 
                {
                    if (contractReferenceNode.Attributes["id"] != null) 
                    {
                        contractReferenceIDRef_ = contractReferenceNode.Attributes["id"].Value;
                        XsdTypeString ob = new XsdTypeString(contractReferenceNode);
                        IDManager.SetID(contractReferenceIDRef_, ob);
                    }
                    else if (contractReferenceNode.Attributes["href"] != null)
                    {
                        contractReferenceIDRef_ = contractReferenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        contractReference_ = new XsdTypeString(contractReferenceNode);
                    }
                }
                else
                {
                    contractReference_ = new XsdTypeString(contractReferenceNode);
                }
            }
            
        
            XmlNode expirationDateNode = xmlNode.SelectSingleNode("expirationDate");
            
            if (expirationDateNode != null)
            {
                if (expirationDateNode.Attributes["href"] != null || expirationDateNode.Attributes["id"] != null) 
                {
                    if (expirationDateNode.Attributes["id"] != null) 
                    {
                        expirationDateIDRef_ = expirationDateNode.Attributes["id"].Value;
                        AdjustableOrRelativeDate ob = new AdjustableOrRelativeDate(expirationDateNode);
                        IDManager.SetID(expirationDateIDRef_, ob);
                    }
                    else if (expirationDateNode.Attributes["href"] != null)
                    {
                        expirationDateIDRef_ = expirationDateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        expirationDate_ = new AdjustableOrRelativeDate(expirationDateNode);
                    }
                }
                else
                {
                    expirationDate_ = new AdjustableOrRelativeDate(expirationDateNode);
                }
            }
            
        
        }
        
    
        #region Multiplier_
        private XsdTypePositiveInteger multiplier_;
        public XsdTypePositiveInteger Multiplier_
        {
            get
            {
                if (this.multiplier_ != null)
                {
                    return this.multiplier_; 
                }
                else if (this.multiplierIDRef_ != null)
                {
                    multiplier_ = IDManager.getID(multiplierIDRef_) as XsdTypePositiveInteger;
                    return this.multiplier_; 
                }
                else
                {
                      return this.multiplier_; 
                }
            }
            set
            {
                if (this.multiplier_ != value)
                {
                    this.multiplier_ = value;
                }
            }
        }
        #endregion
        
        public string multiplierIDRef_ { get; set; }
        #region ContractReference_
        private XsdTypeString contractReference_;
        public XsdTypeString ContractReference_
        {
            get
            {
                if (this.contractReference_ != null)
                {
                    return this.contractReference_; 
                }
                else if (this.contractReferenceIDRef_ != null)
                {
                    contractReference_ = IDManager.getID(contractReferenceIDRef_) as XsdTypeString;
                    return this.contractReference_; 
                }
                else
                {
                      return this.contractReference_; 
                }
            }
            set
            {
                if (this.contractReference_ != value)
                {
                    this.contractReference_ = value;
                }
            }
        }
        #endregion
        
        public string contractReferenceIDRef_ { get; set; }
        #region ExpirationDate_
        private AdjustableOrRelativeDate expirationDate_;
        public AdjustableOrRelativeDate ExpirationDate_
        {
            get
            {
                if (this.expirationDate_ != null)
                {
                    return this.expirationDate_; 
                }
                else if (this.expirationDateIDRef_ != null)
                {
                    expirationDate_ = IDManager.getID(expirationDateIDRef_) as AdjustableOrRelativeDate;
                    return this.expirationDate_; 
                }
                else
                {
                      return this.expirationDate_; 
                }
            }
            set
            {
                if (this.expirationDate_ != value)
                {
                    this.expirationDate_ = value;
                }
            }
        }
        #endregion
        
        public string expirationDateIDRef_ { get; set; }
        
    
        
    
    }
    
}

